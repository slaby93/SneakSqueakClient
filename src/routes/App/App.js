import React from 'react'
import styled from 'styled-components';

export class App extends React.PureComponent {
  render() {
    const { className } = this.props;
    return <span className={className}>App</span>
  }
}

const StyledComponent = styled(App) `
background-color: red;
`;

export default StyledComponent