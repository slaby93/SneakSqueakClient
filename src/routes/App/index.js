import React from 'react';
import AsyncComponent from './../../reactUtils/AsyncComponent'
export default ()=> <AsyncComponent moduleProvider={() => import(/* webpackChunkName: "App" */'./AppContainer')} />;