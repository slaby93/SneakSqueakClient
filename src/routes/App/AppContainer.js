import React from 'react'
import App from './App'
import {connect} from 'react-redux'

const AppContainer = (props) => {
  return <App {...props} />
}

function mapStateToProps (state) {
  const isRequesting = state.getIn(['users', 'isRequesting'])
  const users = state.getIn(['users', 'users'])

  return {
    isRequesting,
    users
  }
}

function mapDispatchToProps (dispatch) {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer)
