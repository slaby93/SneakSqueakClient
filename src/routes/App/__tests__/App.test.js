/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { App } from './../App';

describe('<App/>', () => {
  test('basic render', () => {
    const mountedComponent = shallow(<App />);
    expect(mountedComponent).toMatchSnapshot();
  });
});
