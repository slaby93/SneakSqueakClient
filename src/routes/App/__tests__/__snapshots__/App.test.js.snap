// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> basic render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header__StyledComponent>
          Sneak Squeak
        </Header__StyledComponent>,
        <Button__StyledComponent>
          <Link
            replace={false}
            to="/authorize"
          >
            JOIN US NOW
          </Link>
        </Button__StyledComponent>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Sneak Squeak",
        },
        "ref": null,
        "rendered": "Sneak Squeak",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <Link
            replace={false}
            to="/authorize"
          >
            JOIN US NOW
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "JOIN US NOW",
            "replace": false,
            "to": "/authorize",
          },
          "ref": null,
          "rendered": "JOIN US NOW",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header__StyledComponent>
            Sneak Squeak
          </Header__StyledComponent>,
          <Button__StyledComponent>
            <Link
              replace={false}
              to="/authorize"
            >
              JOIN US NOW
            </Link>
          </Button__StyledComponent>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Sneak Squeak",
          },
          "ref": null,
          "rendered": "Sneak Squeak",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <Link
              replace={false}
              to="/authorize"
            >
              JOIN US NOW
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "JOIN US NOW",
              "replace": false,
              "to": "/authorize",
            },
            "ref": null,
            "rendered": "JOIN US NOW",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
